program/rust/src/tests/test_full_publisher_set.rs

```diff
@@ -0,0 +1,133 @@
+use {
+    crate::{
+        accounts::PriceAccount,
+        c_oracle_header::{
+            PC_NUM_COMP,
+            PC_STATUS_TRADING,
+            PC_STATUS_UNKNOWN,
+        },
+        tests::pyth_simulator::{
+            PythSimulator,
+            Quote,
+        },
+    },
+    solana_sdk::{
+        signature::Keypair,
+        signer::Signer,
+    },
+};
+
+// Verify that the whole publisher set participates in aggregate
+// calculation. This is important for verifying that extra
+// publisher slots on Pythnet are working. Here's how this works:
+//
+// * Fill all publisher slots on a price
+// * Divide the price component array into three parts roughly the
+// same size: first_third, middle_third and last_third
+// * Publish two distinct price values to first/mid third
+// * Verify that the aggregate averages out to an expected value in the middle
+// * Publish PC_STATUS_UNKNOWN prices to first_third to make sure it does not affect later aggregates
+// * Repeat the two-price-value publishing step, using completely
+// different values, this time using mid_third and last_third without first_third
+// * Verify again for the new values meeting in the middle as aggregate.
+#[tokio::test]
+async fn test_full_publisher_set() -> Result<(), Box<dyn std::error::Error>> {
+    let mut sim = PythSimulator::new().await;
+    let pub_keypairs: Vec<_> = (0..PC_NUM_COMP).map(|_idx| Keypair::new()).collect();
```

Comment:
you sure this constant has the right value on each network? (i think you may have added a separate test toggled by the feature flag that it does have the expected value, but i forget)

Reply:
There's a couple potential problems to discuss:
* The size is not computed right - we already explicitly verify in `test_sizes.rs` that Pythnet has 64 and Solana has 32 publishers. 
* In the horrifying case that there's still a mismatch between what the C and Rust code is actually using - it should become evident as part of this test:
** If C code is using 64 and Rust is giving it 32 pubs - we would likely see an invalid memory access crash the program.
** If C code is using 32 and Rust is giving it 64 pubs - there should be an assertion failure due to aggregation working only on the first half of the publisher set, and computing a trivial aggregate price of `100` with confidence `30`, instead of the `110` with conf `20` that's supposed to come out of the two values "meeting in the middle'